### ISSUE TITLE
new Custom generator that supports merging multiple elements from the input

### ISSUE THREAD

**Issue by ymark** [isl/x3ml#69]
Construct a new custom generator that will support the construction of new values by merging multiple similar elements (elements that have a common name). 
Apart from the similar entries, the generator should define the delimiter that will be used for merging the values.

**Comment by ymark** [comment]
The new functionality has been implemented with a new custom generator that can be found under the class gr.forth.ConcatMultipleTerms

The generator can be used for creating both URIs and literal values.

The generator takes at least 3 arguments: 
* **prefix:** it is a constant value and denotes the prefix that should be used before concatenating the terms from the input. If the prefix value starts with *http* then the generated value will be a URI, otherwise, if it is empty or anything else it will generate a literal value.
* **delimiter:** the delimiter that will be used within the concatenated terms
* **text#:** an arbitrary number of terms indicating the XPath expressions that will fetch the terms that will be concatenated. this term text should be followed by an integer indicating the execution order for the concatenation; for example the terms under text1 will be concatenated, then the terms under text2, etc. 

An indicative example of the generator follows. In the generator policy file:
```
<generator name="ConcatMultipleTerms">
   <custom generatorClass="gr.forth.ConcatMultipleTerms">
      <set-arg name="prefix" type="constant"/>
      <set-arg name="delimiter" type="constant"/>
      <set-arg name="text1"/>
      <set-arg name="text2"/>
   </custom>
</generator>
```

and in the mappings.x3ml file:
```
                        <instance_generator name="ConcatMultipleTerms">
							<arg name="prefix" type="constant">http://www.ics.forth.gr/isl/</arg>
   <arg name="text1" type="xpath">ID/text()</arg>
   <arg name="text2" type="xpath">TITLE/text()</arg>
   <arg name="delimiter" type="constant">_</arg>
</instance_generator>
```

**Comment by ymark** [comment]
Apart from the delimiter that is used between different occurrences of the same term, we should also support a different delimiter that will be used when concatenating different terms.

**Comment by ymark** [comment]
Apart from implementing we should guarantee the following:

* if a term does not exist then instead of throwing an error (exception) skip it and continue. This should only be valid for the given type of custom generator (that concatenates similar elements) - A more detailed explanation with examples is given in the next comment.
* if there are spaces in the generated value, and a URI will be generated, then guarantee that it will be generated properly

**Comment by ymark** [comment]
Assuming that we have the following XML input
```
<COIN>
    <ID country="gr">ID_GR_1</ID>
    <ID country="uk">ID_UK_1</ID>
    <TITLE country="gr">Nomisma1</TITLE>
    <TITLE country="uk">Coin1</TITLE>
</COIN> 
<COIN>
    <ID1 country="gr">ID_GR_2</ID1>
    <ID1 country="uk">ID_UK_2</ID1>
</COIN>
```

and the following segment from the mappings file 
```
<domain>
    <source_node>//COIN</source_node>
    <target_node>
        <entity>
            <type>crm:E22_Man-Made_Object</type>
            <instance_generator name="ConcatMultipleTerms">
                <arg name="prefix" type="constant">http://www.ics.forth.gr/isl/</arg>
                <arg name="text1" type="xpath">ID/text()</arg>
                <arg name="text2" type="xpath">TITLE/text()</arg>
                <arg name="sameTermsDelim" type="constant">_</arg>
                <arg name="diffTermsDelim" type="constant">/</arg>
            </instance_generator>
        </entity>
    </target_node>
</domain>
```

The engine should create the following URIS:

* http://www.ics.forth.gr/isl/ID_GR_1_ID_UK_1/Nomisma1_Coin1
* http://www.ics.forth.gr/isl/ID_GR_2_ID_UK_2

The only case where a value should not be generated is, if all terms (i.e. text1, text2) cannot be found in the input.

**Comment by ymark** [comment]
Th functionality of the custom generator has been updated. 
The new generator will be available in release 1.7.4
